csharp/src/Ice/ObjectAdapter.cs

```diff
@@ -760,43 +756,41 @@ public void RefreshPublishedEndpoints()
             }
         }
 
-        /// <summary>Retrieves a copy of the endpoints that would be listed in a proxy created by this object adapter.
+        /// <summary>Retrieves the endpoints that would be listed in a proxy created by this object adapter.
         /// </summary>
         /// <returns>The published endpoints.</returns>
-        public Endpoint[] GetPublishedEndpoints()
+        public IReadOnlyList<Endpoint> GetPublishedEndpoints()
         {
             lock (_mutex)
             {
-                return (Endpoint[])_publishedEndpoints.Clone();
+                return _publishedEndpoints;
             }
         }
 
         /// <summary>Sets the endpoints that from now on will be listed in the proxies created by this object adapter.
         /// </summary>
         /// <param name="newEndpoints">The new published endpoints.</param>
-        public void SetPublishedEndpoints(Endpoint[] newEndpoints)
+        public void SetPublishedEndpoints(IEnumerable<Endpoint> newEndpoints)
         {
             LocatorInfo? locatorInfo = null;
-            Endpoint[] oldPublishedEndpoints;
+            IReadOnlyList<Endpoint> oldPublishedEndpoints;
 
             lock (_mutex)
             {
                 CheckForDeactivation();
                 if (_routerInfo != null)
                 {
                     throw new InvalidOperationException(
-                                    "cannot set published endpoints on an object adapter associated with a router");
+                        "cannot set published endpoints on an object adapter associated with a router");
                 }
 
                 oldPublishedEndpoints = _publishedEndpoints;
-                if (newEndpoints.Length == 0)
+                _publishedEndpoints = newEndpoints.ToArray();
+                if (_publishedEndpoints.Count == 0)
                 {
+                    // TODO: is it correct to set published endpoints to empty?
```

Comment:
Although probably uncommon, I'd rather not be too restrictive by disallowing it. A location mechanism might not necessarily used the configured published endpoints or locating OAs.